### PR TITLE
Device mode improvements for RN 2.0

### PR DESCRIPTION
**Pull request by niallzato**
### Proposed changes
Clarify device/cloud mode support for RN 2.0, and link to correct docs for Firebase.

### CODE REVIEW

File: src/connections/sources/catalog/libraries/mobile/react-native/index.md [hunk 1]
```diff
@@ -492,7 +492,7 @@ These are the example plugins you can use and alter to meet your tracking needs:
 | IDFA                | `@segment/analytics-react-native-plugin-idfa`                |
 
 ## Supported Destinations
-Segment supports these destinations for Analytics React Native 2.0 in device-mode, with more to follow. Cloud-mode destinations are also supported:
+Segemnt supports a large number of [Cloud-mode](https://segment.com/docs/connections/destinations/#connection-modes) destinations. Segment also supports the below destinations for Analytics React Native 2.0 in device-mode, with more to follow:
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/react-native/index.md:495]:
```suggestion
Segment supports a large number of [Cloud-mode](https://segment.com/docs/connections/destinations/#connection-modes) destinations. Segment also supports the below destinations for Analytics React Native 2.0 in device-mode, with more to follow:
```

Comment (stayseesong) [reply src/connections/sources/catalog/libraries/mobile/react-native/index.md:495]: @niallzato what does "large number" of cloud mode destinations mean? Does it support all cloud-mode destinations? Is there a list?

Comment (niallzato) [reply src/connections/sources/catalog/libraries/mobile/react-native/index.md:495]: Hey @stayseesong  good question! We don't have a list of cloud mode integrations that work with mobile. I would guess it's compatible with all cloud mode destinations, but considering how big our catalog is it's possible certain destinations do not send mobile data in cloud mode. That was the case I was trying to cover for. 

Happy to change it to `all`, I was really just trying to cover a possible edge case.

**Comment by niallzato** [comment]
Hey @stayseesong just giving this a bump :)

File: src/connections/destinations/catalog/firebase/index.md [hunk 2]
```diff
@@ -108,14 +108,15 @@ By default, Segment only bundles `Firebase/Core` which is [Firebase's Analytics
...
-If you use Segment's React Native source library, you must explicitly bundle the mobile SDKs for both iOS and Android with your project.
+If you use Segment's older React Native source library, you must explicitly bundle the mobile SDKs for both iOS and Android with your project.
 
 1. Use yarn to add the `analytics-react-native-firebase` SDKs.  (`@segment/analytics-react-native-firebase`)
 2. Add `import` statements to your code so you can access the SDKs.
 3. Go to the `await analytics.setup` configuration in your code, and find (or add) the `using:` item. Add "Firebase" to the list of device-mode destinations in the `using` item.
 4. Change to your iOS directory and run `pod install`.
 5. Add the `analytics-react-native-firebase` module to your `build.gradle` file. (See Step.4 of [Getting Started on Android](/docs/connections/destinations/catalog/firebase/#getting-started-on-android))
 
+For React Native 2.0 you can reference the [install guide](https://github.com/segmentio/analytics-react-native/blob/master/packages/plugins/plugin-firebase/README.md).
```
Comment (stayseesong) [inline src/connections/destinations/catalog/firebase/index.md:119]:
```suggestion
For React Native 2.0 you can reference the [install guide](https://github.com/segmentio/analytics-react-native/blob/master/packages/plugins/plugin-firebase/README.md){:target="_blank"}.
```

File: src/connections/sources/catalog/libraries/mobile/react-native/index.md [hunk 3]
```diff
@@ -492,7 +492,7 @@ These are the example plugins you can use and alter to meet your tracking needs:
 | IDFA                | `@segment/analytics-react-native-plugin-idfa`                |
 
 ## Supported Destinations
-Segment supports these destinations for Analytics React Native 2.0 in device-mode, with more to follow. Cloud-mode destinations are also supported:
+Segment supports a large number of [Cloud-mode](https://segment.com/docs/connections/destinations/#connection-modes) destinations. Segment also supports the below destinations for Analytics React Native 2.0 in device-mode, with more to follow:
```
Comment (stayseesong) [inline src/connections/sources/catalog/libraries/mobile/react-native/index.md:495]:
```suggestion
Segment supports a large number of [Cloud-mode](/docs/connections/destinations/#connection-modes) destinations. Segment also supports the below destinations for Analytics React Native 2.0 in device-mode, with more to follow:
```

**Review by stayseesong — APPROVED** [review]

**Pull request merged**